### DISCUSSION TITLE
Multiple formatters with List

### DISCUSSION THREAD

**Discussion by simplecommerce** [i18next/i18next#1752] [Q&A] [answered]
Hi,

I was wondering if there was a simple way to do the following.

I have an array which I am passing to the `list` built-in formatter.
I would like to be able to apply another format but not to the list itself but to each individual item of the list.

Right now the only way I was able to achieve it, was to add a custom formatter before my `list` that maps my array into the list of strings that I then pass to list.

But I thought it would be nice to be able to do the following.

```
i18next.t("translation","{{val, list, customFormat}}");
```

And it could be anyting from the built-in ones to custom ones.

For example, I want to format my array of numbers using list and currency.

How do you guys achieve this or do you have suggestions?

**Comment by jamuhl** [comment, marked as answer]
Can‘t you just add a customformatter with unique name like described here https://www.i18next.com/translation-function/formatting#basic-usage

**Reply by simplecommerce** [reply]
> Can‘t you just add a customformatter with unique name like described here https://www.i18next.com/translation-function/formatting#basic-usage

Maybe, I just don't see how I could achieve that with my example.

If I have an array of numbers that I would want to use list and format it into an array of currency so it would end up being:

`$10,000, $20,000 and $30,000`  when I am passing `[10000,20000,30000]`

or an array of dates.

I read the doc but I just don't see how I could do it without creating a custom one.

**Reply by jamuhl** [reply]
like said by adding a formatter that loops through value (array) to map items to currency strings and finally do the intl list stuff

**Reply by simplecommerce** [reply]
> like said by adding a formatter that loops through value (array) to map items to currency strings and finally do the intl list stuff

yeah that's what I ended up doing, I just thought maybe there was a better way to achieve this.

I know its not part of this discussion but I don't understand what `alwaysFormat` does, since there isn't an example in the docs of what it can be use for, like a use case.

**Reply by jamuhl** [reply]
not sure how you think that can be simplified? but open for suggestions. 

regarding alwaysFormat I have no idea. @adrai might know more?

**Comment by adrai** [comment]
alwaysFormat, will always format an interpolated value...
example: https://github.com/i18next/i18next/blob/master/test/interpolation.spec.js#L269

Introduced with https://github.com/i18next/i18next/pull/1385 based on issue https://github.com/i18next/i18next/issues/1384 by @mjbcopland

**Reply by simplecommerce** [reply]
> alwaysFormat, will always format an interpolated value... example: https://github.com/i18next/i18next/blob/master/test/interpolation.spec.js#L269
> 
> Introduced with #1385 based on issue #1384 by @mjbcopland

Ah ok, so from what I understand, in the test case, it will always execute the format function and return the lowercase result when no formatting is passed, this is the only use case I gather.

**Reply by adrai** [reply]
yes
